staging/src/k8s.io/client-go/tools/clientcmd/api/types.go

```diff
@@ -67,6 +67,9 @@ type Cluster struct {
 	LocationOfOrigin string
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
+	// ServerName is used to check server ceritificates. If ServerName is empty, the hostname used to contact the server is used.
+	// +optional
+	ServerName string `json:"server-name,omitempty"`
```

Comment:
same comment about `tls-server-name` or `tls-verify-server-name` to avoid confusion